src/shims/env.rs

```diff
@@ -92,6 +115,35 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         }
     }
 
+    fn setenvironmentvariablew(
+        &mut self,
+        name_op: OpTy<'tcx, Tag>,
+        value_op: OpTy<'tcx, Tag>,
+    ) -> InterpResult<'tcx, i32> {
+        let this = self.eval_context_mut();
+
+        let name_ptr = this.read_scalar(name_op)?.not_undef()?;
+        let value_ptr = this.read_scalar(value_op)?.not_undef()?;
+        let value = this.memory.read_wide_str(value_ptr)?;
+        let mut new = None;
+        if !this.is_null(name_ptr)? {
+            let name = this.memory.read_wide_str(name_ptr)?;
+            if !name.is_empty() && !name.contains(&b'=') {
```

Comment:
E.g. here you are mixing up encodings. You can't do a byte-wise search for the ASCII `=` in a UTF-16 encoded string.